packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -667,6 +671,22 @@ See documentation: ${createDocumentationLink({
     this.middleware.forEach(({ instance }) => {
       instance.started();
     });
+
+    // This is the automatic Insights middleware,
+    // added when `insights` is unset and the initial results possess `queryID`.
+    // Any user-provided middleware will be added later and override this one.
+    if (typeof this._insights === 'undefined') {
+      mainHelper.derivedHelpers[0].once('result', () => {
+        const hasQueryID = this.mainIndex
+          .getScopedResults()
+          .some(({ results }) => results?.queryID !== undefined);
```

Comment:
we could need to check that clickAnalytics: true wasn't in the parameters

(and rules / secured api key, but that's not readable easily, maybe via `params`)

Reply:
This is not needed anymore, as we have an explicit option coming from the engine now.